### ISSUE TITLE
The label of synthesized data and real data

### ISSUE THREAD

**Issue by contstriver** [jason718/game-feature-learning#2]
Hello jason, thanks for your hard work.
It seems that the synthesized data is assigned to 0 in the code and real data is set to 1, which is different from the paper.
 ``self.loss_D_syn = self.criterionGAN(pred_syn, False)``
 ``  self.loss_D_real = self.criterionGAN(pred_real, True) ``
I'm new to the field of domain adaption and GAN.  I don't know if this will affect the performance.
Looking forward your response. Thank you.

**Comment by jason718** [comment]
Thanks for comment.  It's a binary cross entropy (BCE) loss, so it will make no difference in theory.

**Comment by contstriver** [comment]
Thanks for your answer. I have another simple question about the choice of domain transfer model. 

I think your proposed model can be viewed as a kind of pipline which shares the weight used in the paper **Domain adversarial training of neural networks**.  There is also another kind of model that doesn't share the weight like in **Adversarial discriminative domain adaptation**. 

I  also implement the second method and find that it hard to train. I want to know if you have ever tried to use this method before.  If so, how about the result ?

**Comment by jason718** [comment]
I agreed, plus I didn't try to implement these two papers. But I have seen other papers using it and managed to make it work. (https://arxiv.org/pdf/1803.03243.pdf ...)

I think whether your domain adaptation module working or not depends on lots of things. In my case, because of the domain difference (indoor synthetic -> any real-world images) is very big, it's also very very hard to make it work. 

Maybe you should contact the authors of the second paper for instructions.

**Comment by contstriver** [comment]
I have decided to get rid of the second paper. :)

Also, I find the hyperparameter will make a huge impact on the performance like the learning rate of discriminator/generator and weight of different loss in gloss. Do you also experience the same situation? 

Meanwhile, I have another question that how to decide to end the training. In other words, how to know if the target model is good enough when you don't have the target label to test it.  In my case, I first use a target data set  with  label to test the model, and I find the test loss curve is always rise down and up. And after several epochs, the test loss will keep rising. This makes me  hard to decide when to stop the training where there is no ground truth of target data set. 

It seems that the epoch in DA is relative small compare to other application. And I find that when the DA_WEIGHT in gloss is  small can achieve a more stable result in my case.

Thanks for your patience.

**Comment by jason718** [comment]
yes, I ended up setting the adv. loss weight to be pretty small. 

In my case, I also don't have label on target domain. But I can test on other tasks on target domain, which are of my interest. I just keep evaluating on it. I think you are right, if you want to joint train everything, setting the weight to be a relatively small value helps.
